### DISCUSSION TITLE
Is there any way to calculate how much native memory the onnxruntime will need for a model?

### DISCUSSION THREAD

**Discussion by tmbrye** [microsoft/onnxruntime#21739] [Other Q&A]
We are using the Java wrapper for the onnxruntime and running in an environment where we would like to have some guard rails as to how much native memory the onnxruntime uses.  Is there any way of calculating based on the onnx model how much native memory it will need to allocate or maybe some type of expansion factor guideline based on the input models size?

Thanks.

**Comment by tianleiwu** [comment]
No currently.
It depends on model. You can use some test data to run the model, and monitor the memory usage. For example, change the allocator expand strategy to SameAsRequest, and warm up with some input that will need most memory. Then you know the upper bound of memory usage.
